test/Microsoft.Extensions.Localization.Tests/ResourceManagerStringLocalizerFactoryTest.cs

```diff
@@ -109,5 +217,48 @@ public void Create_FromNameLocation_ReturnsNewResultForDifferentLocation()
             // Assert
             Assert.NotSame(result1, result2);
         }
+
+        [Fact]
+        public void Create_FromNameLocation_ResourcesPathDirectorySeperatorToDot()
+        {
+            // Arrange
+            var hostingEnvironment = new Mock<IHostingEnvironment>();
+            hostingEnvironment.SetupGet(a => a.ApplicationName).Returns("Microsoft.Extensions.Localization.Tests");
+            var locOptions = new LocalizationOptions();
+            locOptions.ResourcesPath = Path.Combine("My", "Resources");
+            var options = new Mock<IOptions<LocalizationOptions>>();
+            options.Setup(o => o.Value).Returns(locOptions);
+            var factory = new TestResourceManagerStringLocalizerFactory(
+                hostingEnvironment.Object,
+                options.Object,
+                resourceLocationAttribute: null);
+
+            // Act
+            var result1 = factory.Create("baseName", location: null);
+
+            // Assert
+            Assert.DoesNotContain(Path.DirectorySeparatorChar, factory.BaseName);
```

Comment:
Look for the actual expected resource path to match the test name. Should not be OS-sensitive.